packages/monitors/src/OptimisticOracleContractMonitor.js

```diff
@@ -85,21 +88,32 @@ class OptimisticOracleContractMonitor {
     latestEvents = latestEvents.filter((event) => event.blockNumber > this.lastRequestPriceBlockNumber);
 
     for (let event of latestEvents) {
-      const convertCollateralDecimals = await this._getCollateralDecimalsConverted(event.currency);
+      const convertCollateralDecimals = await this._getCollateralDecimalsConverted(
+        this.oracleType === OptimisticOracleType.OptimisticOracle ? event.currency : event.request.currency
+      );
       const mrkdwn =
         createEtherscanLinkMarkdown(event.requester, this.contractProps.networkId) +
         ` requested a price at the timestamp ${event.timestamp} for the identifier: ${event.identifier}. ` +
         `The ancillary data field is ${event.ancillaryData || "0x"}. ` +
-        `Collateral currency address is ${event.currency}. Reward amount is ${this.formatDecimalString(
-          convertCollateralDecimals(event.reward)
-        )} and the final fee is ${this.formatDecimalString(convertCollateralDecimals(event.finalFee))}. ` +
+        `Collateral currency address is ${
+          this.oracleType === OptimisticOracleType.OptimisticOracle ? event.currency : event.request.currency
```

Comment:
o boy this is such a pain in the ass. thanks for sorting out this switching logic!

Reply:
lol its ugly